flash/text/seq2seq/core/model.py

```diff
@@ -0,0 +1,148 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import warnings
+from typing import Any, Callable, List, Mapping, Optional, Sequence, Type, Union
+
+import pytorch_lightning as pl
+import torch
+from pytorch_lightning.utilities import rank_zero_info
+from transformers import AutoModelForSeq2SeqLM
+
+from flash.core import Task
+
+
+def _pad_tensors_to_max_len(model_cfg, tensor, max_length):
+    pad_token_id = model_cfg.pad_token_id if model_cfg.pad_token_id is not None else model_cfg.eos_token_id
+    if pad_token_id is None:
+        raise ValueError(
+            f"Make sure that either `config.pad_token_id` or `config.eos_token_id` "
+            f"is defined if tensor has to be padded to `max_length`={max_length}"
+        )
+
+    padded_tensor = pad_token_id * torch.ones((tensor.shape[0], max_length), dtype=tensor.dtype, device=tensor.device)
+    padded_tensor[:, :tensor.shape[-1]] = tensor
+    return padded_tensor
+
+
+class Seq2SeqTask(Task):
+    """General Task for Sequence2Sequence.
+    Args:
+        loss_fn: Loss function for training
+        optimizer: Optimizer to use for training, defaults to `torch.optim.Adam`.
+        metrics: Metrics to compute for training and evaluation.
+        learning_rate: Learning rate to use for training, defaults to `3e-4`
+        val_target_max_length: Maximum length of targets in validation. Defaults to `128`
+        num_beams: Number of beams to use in validation when generating predictions. Defaults to `4`
+        freeze_embeds: Freeze Embedding layers within the backbone. Defaults to `True`
+    """
+
+    def __init__(
+        self,
+        backbone: str = 't5-small',
+        loss_fn: Optional[Union[Callable, Mapping, Sequence]] = None,
+        optimizer: Type[torch.optim.Optimizer] = torch.optim.Adam,
+        metrics: Union[pl.metrics.Metric, Mapping, Sequence, None] = None,
+        learning_rate: float = 5e-5,
+        val_target_max_length: Optional[int] = None,
+        num_beams: Optional[int] = None,
+        freeze_embeds: bool = True
+    ):
+        os.environ["TOKENIZERS_PARALLELISM"] = "TRUE"
+        # disable HF thousand warnings
+        warnings.simplefilter("ignore")
```

Comment:
shall it be back enabled on object destroy, in `__del__`

Reply:
You mean reset this back to False? I think it's fine for now, we can decide later (it's also an issue in text classification)

Reply:
then I would move it to upper for example to init

Reply:
Ah but that will add clutter to the example, I think it's best the logic is resolved within a base class (that's how it will be handled eventually!). Definitely something important, but not too big an issue to deal with now I think!